arlas-core/src/main/java/io/arlas/server/services/ExploreServices.java

```diff
@@ -394,71 +406,101 @@ public AggregationResponse formatAggregationResult(MultiBucketsAggregation aggre
+                            for (SearchHit hit: hits) {
+                                Map source = hit.getSourceAsMap();
+                                if (rawGeometries != null && !CollectionUtils.isEmpty(rawGeometries.geometries)) {
+                                    rawGeometries.geometries.forEach(g -> {
+                                        GeoJsonObject geometryGeoJson = null;
+                                        try {
+                                            CollectionReferenceManager.setCollectionGeometriesType(source, collection, rawGeometries.geometries.stream().collect(Collectors.joining(",")));
+                                            GeoTypeEnum geometryType = null;
+                                            Object geometry = MapExplorer.getObjectFromPath(g, source);
+                                            ReturnedGeometry returnedGeometry = null;
+                                            if (geometry != null) {
+                                                returnedGeometry = new ReturnedGeometry();
+                                                geometryType = collection.params.getGeometryType(g);
+                                                returnedGeometry.reference = g;
+                                            } else {
+                                                geometry = MapExplorer.getObjectFromPath(collection.params.centroidPath, source);
+                                                geometryType = collection.params.getGeometryType(collection.params.centroidPath);
+                                                if (geometry != null) {
+                                                    returnedGeometry = new ReturnedGeometry();
+                                                    returnedGeometry.reference = collection.params.centroidPath;
+                                                }
+                                            }
+                                            geometryGeoJson = geometry != null ?
+                                                    GeoTypeMapper.getGeoJsonObject(geometry, geometryType) : null;
+                                            if (geometryGeoJson != null) {
+                                                returnedGeometry.geometry = geometryGeoJson;
+                                                returnedGeometry.isRaw = true;
+                                                if (element.geometries == null) {
+                                                    element.geometries = new ArrayList<>();
+                                                }
+                                                element.geometries.add(returnedGeometry);
+                                            }
+                                        } catch (ArlasException e) {
+                                            e.printStackTrace();
```

Comment:
Won't the exception be hidden?

Reply:
Yes totally. This is a hack to keep rendering the geojson even when we can't parse an unkown format of the indexed geometry.

It should be treated in another PR 

I'll create the issue